lib/matplotlib/backend_tools.py

```diff
@@ -266,24 +266,27 @@ def _add_tool_cbk(self, event):
         self._add_tool(event.tool)
 
     def _set_cursor_cbk(self, event):
-        if not event:
+        if not event or not self.canvas:
             return
         if (self._current_tool and getattr(event, "inaxes", None)
                 and event.inaxes.get_navigate()):
             if self._last_cursor != self._current_tool.cursor:
-                self.set_cursor(self._current_tool.cursor)
+                self.canvas.set_cursor(self._current_tool.cursor)
                 self._last_cursor = self._current_tool.cursor
         elif self._last_cursor != self._default_cursor:
-            self.set_cursor(self._default_cursor)
+            self.canvas.set_cursor(self._default_cursor)
             self._last_cursor = self._default_cursor
 
+    @_api.deprecated("3.5", alternative="figure.canvas.set_cursor")
     def set_cursor(self, cursor):
         """
         Set the cursor.
-
-        This method has to be implemented per backend.
         """
-        raise NotImplementedError
+        self.canvas.set_cursor(cursor)
+
+
+# This exists solely for deprecation warnings.
```

Comment:
Can you add some context on when this can be removed? Otherwise, we'll always be confused if this is still relevant or not.